src/middlewared/middlewared/plugins/replication.py

```diff
@@ -1,393 +1,440 @@
-            Int('remote_port', default=22, required=True),
-            Str('begin', validators=[Time()]),
-            Str('compression', enum=['OFF', 'LZ4', 'PIGZ', 'PLZIP']),
-            Str('end', validators=[Time()]),
-            Str('filesystem', required=True),
-            Str('remote_cipher', enum=['STANDARD', 'FAST', 'DISABLED']),
-            Str('remote_dedicateduser'),
-            Str('remote_hostkey'),
-            Str('remote_hostname', required=True),
-            Str('remote_mode', enum=['SEMIAUTOMATIC', 'MANUAL'], required=True),
-            Str('remote_token'),
-            Str('zfs', required=True),
-            register=True
+            "replication_create",
+            Str("direction", enum=["PUSH", "PULL"], required=True),
+            Str("transport", enum=["SSH", "SSH+NETCAT", "LOCAL", "LEGACY"], required=True),
+            Int("ssh_credentials", null=True, default=None),
+            Str("netcat_active_side", enum=["LOCAL", "REMOTE"], null=True, default=None),
+            Int("netcat_active_side_port_min", null=True, default=None, validators=[Port()]),
+            Int("netcat_active_side_port_max", null=True, default=None, validators=[Port()]),
+            List("source_datasets", items=[Path("dataset", empty=False)], required=True, empty=False),
+            Path("target_dataset", required=True, empty=False),
+            Bool("recursive", required=True),
+            List("exclude", items=[Path("dataset", empty=False)], default=[]),
+            List("periodic_snapshot_tasks", items=[Int("periodic_snapshot_task")], default=[],
+                 validators=[Unique()]),
+            List("naming_schema", items=[
+                Str("naming_schema", validators=[ReplicationSnapshotNamingSchema()])], default=[]),
+            List("also_include_naming_schema", items=[
+                Str("naming_schema", validators=[ReplicationSnapshotNamingSchema()])], default=[]),
+            Bool("auto", required=True),
+            Cron("schedule", begin_end=True, null=True, default=None),
+            Cron("restrict_schedule", begin_end=True, null=True, default=None),
+            Bool("only_matching_schedule", default=False),
+            Bool("allow_from_scratch", default=False),
+            Bool("hold_pending_snapshots", default=False),
+            Str("retention_policy", enum=["SOURCE", "CUSTOM", "NONE"], required=True),
+            Int("lifetime_value", null=True, default=None, validators=[Range(min=1)]),
+            Str("lifetime_unit", null=True, default=None, enum=["HOUR", "DAY", "WEEK", "MONTH", "YEAR"]),
+            Str("compression", enum=["LZ4", "PIGZ", "PLZIP"], null=True, default=None),
+            Int("speed_limit", null=True, default=None, validators=[Range(min=1)]),
+            Bool("dedup", default=True),
+            Bool("large_block", default=True),
+            Bool("embed", default=True),
+            Bool("compressed", default=True),
```

Comment:
Hmm, would it make it simpler to have this be a list with enabled features? `enable_features = ["DEDUP", "LARGE_BLOCK", "EMBED", "COMPRESSED"]`.

Not strong about it, just a thought.